content-fragment-workflows/src/web-src/src/components/Workflows/ContentFragmentsSelection.js

```diff
@@ -0,0 +1,251 @@
+      const contentFragments = await getContentFragments({
+        aemHost,
+        token,
+        fragmentIds,
+      });
+      const fragmentPathsToProceedReferences = contentFragments.filter((item) => item.references.length > 0).map((item) => item.path);
+      const fragmentPathsToProceedReferencedBy = contentFragments.filter((item) => item.referencedBy.length > 0).map((item) => item.path);
+
+      // for table view
+      setContentFragments(contentFragments);
+      setSelectedReferencesValues(fragmentPathsToProceedReferences);
+      setSelectedReferencedByValues(fragmentPathsToProceedReferencedBy);
+
+      // for workflow processing within the parent component
+      const contentFragmentsToProceed = getUniqueContentFragments(
+          contentFragments,
+          fragmentPathsToProceedReferences,
+          fragmentPathsToProceedReferencedBy
+      );
+      setContentFragmentsToProceed(contentFragmentsToProceed);
+
+      setIsErrorWithContentFragments(false);
+      setIsLoadingContentFragments(false);
+    };
+
+    init().catch((e) => {
+      console.log("Extension got the error during load content fragments:", e);
+      setIsErrorWithContentFragments(true);
+      setIsLoadingContentFragments(false);
+    });
+  }, [aemHost, token, fragmentIds.join(',')]);
+
+  const onChangeReferencesHandler = (value, isSelected) => {
+    const updatedSelectedReferencesValues = isSelected
+        ? [...selectedReferencesValues, value]
+        : selectedReferencesValues.filter(item => item !== value);
+
+    // for table view
+    setSelectedReferencesValues(updatedSelectedReferencesValues);
+    // for workflow processing within the parent component
+    const contentFragmentsToProceed = getUniqueContentFragments(
+        contentFragments,
+        updatedSelectedReferencesValues,
+        selectedReferencedByValues
+    );
+    setContentFragmentsToProceed(contentFragmentsToProceed);
+  };
+
+  const onChangeReferencedByHandler = (value, isSelected) => {
+    const updatedSelectedReferencedByValues = isSelected
+        ? [...selectedReferencedByValues, value]
+        : selectedReferencedByValues.filter(item => item !== value);
+
+    // for table view
+    setSelectedReferencedByValues(updatedSelectedReferencedByValues);
+    // for workflow processing within the parent component
+    const contentFragmentsToProceed = getUniqueContentFragments(
+        contentFragments,
+        selectedReferencesValues,
+        updatedSelectedReferencedByValues
+    );
+    setContentFragmentsToProceed(contentFragmentsToProceed);
+  };
+
+  if (isLoadingContentFragments === true) {
+    return <LoadingContentFragments />;
+  } else if (isErrorWithContentFragments === true) {
+    return <FetchError aemHost={aemHost} />;
+  } else {
+    const columns = [
+      { name: "Title", key: "title", width: "3fr" },
+      { name: "Status", key: "status", width: "1fr" },
+      { name: "References", key: "references", width: "1fr" },
```

Comment:
Need to add a tooltip with some explanation "what is References and Referenced By"